lib/swaggerUtil.js

```diff
@@ -24,6 +24,106 @@ exports.validateJSONType = function (schema, value) {
     return result;
 };
 
+exports.getObjectsWithDiscriminator = function (object) {
+    var map = {};
+    var properties = object.properties;
+    var discriminator = '';
+    if (_.has(object, 'allOf') || _.has(object, 'items.allOf')) {
+        var inheritedModels = object.allOf ? object.allOf : object.items.allOf;
+        map.isArray = object.type === 'array' ? true : undefined;
+        inheritedModels.forEach(function (inheritedModel) {
+            _.merge(map, exports.getObjectsWithDiscriminator(inheritedModel));
+        });
+    }
+    if (_.has(object, 'discriminator') || _.has(object, 'items.discriminator')) {
+        if (object.type === 'array') {
+            properties = object.items.properties;
+            discriminator = object.items.discriminator;
+            map.isArray = true;
+        }
+        else {
+            discriminator = object.discriminator;
+        }
+        map.discriminator = discriminator;
+    }
+    var mapProp;
+    _.forOwn(properties, function (value, key) {
+        if (value.type === 'object' || value.type === 'array' || value.allOf) { //is this a complex type
+            mapProp = exports.getObjectsWithDiscriminator(value);
+            if (Object.keys(mapProp).length != 0) { //avoid adding property to map if it is an empty object
+                map[key] = mapProp;
+            }
+        }
+    });
+    return map;
+};
+
+exports.validateIndividualObjects = function (specs, map, data) {
+    var validationErrors = [];
+    if (map.discriminator) {
+        if (map.isArray) {
+            data.forEach(function (item) {
+                validationErrors = validationErrors.concat(validateObject(specs, item, map.discriminator));
+            });
+        }
+        else {
+            validationErrors = validationErrors.concat(validateObject(specs, data, map.discriminator));
+        }
+    }
+    var errors;
+    if (Array.isArray(data)) {
+        for (var i=0;i<data.length;i++) {
+            errors = loopThroughMapProps(specs, map, data[i]);
+            if (errors === 'no nested objects') {
```

Comment:
this return value and the check for it looks error prone ... should not be checking for a specific string like this